src/reducers/membersNew.ts

```diff
@@ -295,12 +330,33 @@ function applyOperation(
         const { amount } = operation.data;
 
         assertUidPresentInState(prevState, creator_uid, operation);
-        const minter = prevState.byUid[creator_uid].mint(
+        const minter = prevState.byUid[creator_uid].mintRaha(
           new Big(amount),
           new Date(operation.created_at)
         );
         return addMembersToState(prevState, [minter]);
       }
+      case OperationType.GIVE: {
+        const { to_uid, amount, donation_to, donation_amount } = operation.data;
+
+        assertUidPresentInState(prevState, creator_uid, operation);
+        assertUidPresentInState(prevState, to_uid, operation);
+        assertUidPresentInState(prevState, donation_to, operation);
+
+        const giver = prevState.byUid[creator_uid].giveRaha(
+          new Big(amount).plus(donation_amount)
+        );
+        const receiver = prevState.byUid[to_uid].receiveRaha(new Big(amount));
+        const donationRecipient = prevState.byUid[donation_to].receiveRaha(
+          new Big(donation_amount)
```

Comment:
Curious, how does this respond to non numeric strings?

Reply:
(Note this commit is part of the Give component PR, as I didn't want to have to deal with rebasing issues.